src/FunctionApps/DSTP/src/main/kotlin/gov/cdc/prime/phdi/ParseHL7.kt

```diff
@@ -0,0 +1,66 @@
+package gov.cdc.prime.phdi
+
+import java.util.*
+import com.microsoft.azure.functions.*
+import com.microsoft.azure.functions.annotation.*
+
+import gov.cdc.prime.phdi.Utilities.HL7Reader
+import gov.cdc.prime.phdi.Utilities.HL7Parser
+import gov.cdc.prime.phdi.Utilities.HL7Validator
+
+class ParseHL7 {
+    @FunctionName("parsehl7")
+    fun run(
+        @BlobTrigger(
+            name = "file",
+            dataType = "binary",
+            path = "bronze/decrypted/{name}",
+            connection="AzureWebJobsStorage"
+        ) content: ByteArray,
+        @BindingName("name") filename: String,
+        @BlobOutput(
+            name="validTarget",
+            dataType = "string",
+            path="bronze/validation/{name}.valid",
+            connection="AzureWebJobsStorage"
+        ) validContent: OutputBinding<String>,
+        @BlobOutput(
+            name="invalidTarget",
+            dataType = "string",
+            path="bronze/validation/{name}.invalid",
+            connection="AzureWebJobsStorage"
+        ) invalidContent: OutputBinding<String>,
+        context: ExecutionContext
+	) {
+        context.logger.info("Name: ${filename} Size: ${content.size} bytes.")
+
+        // we don't actually do any processing of the HL7 messages here
+        // We read in files from blob storage, extract individual messages from that
+        // file, and then check that each message is valid. If it is, it goes to one
+        // bucket, and if it's not it goes to another.
+
+        val reader = HL7Reader()
+        val parser = HL7Parser()
+        val validator = HL7Validator()
+        val cleanedMessages: String = reader.readHL7MessagesFromByteArray(content)
```

Comment:
"cleaned" implies that there's some processing other than byte conversion here -- can we find another variable name?